src/libsync/iconjob.h

```diff
@@ -27,14 +29,14 @@ namespace OCC {
  * @brief Job to fetch a icon
  * @ingroup gui
  */
-class IconJob : public QObject
+class OWNCLOUDSYNC_EXPORT IconJob : public QObject
 {
     Q_OBJECT
 public:
     explicit IconJob(const QUrl &url, QObject *parent = nullptr);
 
 signals:
-    void jobFinished(QByteArray iconData);
+    void jobFinished(const QByteArray &iconData);
```

Comment:
@FlexW Won't we have a bug if `reply` gets deleted somewhere in the middle and we are returning a `const reference` to `QByteArray`? In fact, `QByteArray` is using Implicit Sharing which should be safe enough and the copy is not created if not needed.

Reply:
Okay, that makes sense.